### PR TITLE
Utilize check_batch_axis in Model._standardize_user_data

### PR DESCRIPTION
**Pull request by sadreamer**
Previous PR, problem definition and discussion: https://github.com/keras-team/keras/pull/9110

In short, the problem is that `check_batch_axis` is not used within `Model._standardize_user_data` and you can not easily disable the validation of input data's batch axis (when you really need it).

### CODE REVIEW

**Comment by sadreamer** [comment]
@fchollet Done. I also realized that the semantics of `_standardize_input_data.check_batch_axis` differs from `_standardize_user_data.check_batch_axis`. In `_standardize_input_data`, we check whether input shapes coincide with the expected ones (which are set in `InputLayer`s in particular). In `_standardize_user_data`, we check whether all inputs have the same values of `.shape[0]`.
So in order to reduce this ambiguity, in the last commit I also made corresponding renamings of functions/arguments.

File: keras/engine/training.py [hunk 1]
```diff
@@ -180,8 +180,8 @@ def _standardize_sample_weights(sample_weight, output_names):
                                                 'sample_weight')
 
 
-def _check_array_lengths(inputs, targets, weights=None):
-    """Does user input validation for numpy arrays.
+def _check_batch_axes_coincide(inputs, targets, weights=None):
```
Comment (fchollet) [inline keras/engine/training.py]: `_check_array_lengths` was arguably a better name

File: keras/engine/training.py [hunk 2]
```diff
@@ -1399,9 +1399,9 @@ def _test_loop(self, f, ins, batch_size=None, verbose=0, steps=None):
             return outs[0]
         return outs
 
-    def _standardize_user_data(self, x, y,
-                               sample_weight=None, class_weight=None,
-                               check_batch_axis=True, batch_size=None):
+    def _standardize_training_data(self, x, y,
```
Comment (fchollet) [inline keras/engine/training.py]: This is not training data, since it is used in `evaluate`/etc. Proper name is `_standardize_user_data`

**Comment by sadreamer** [comment]
Fixed.
